userspace/libscap/win32/scap_machine_info.c

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright (C) 2023 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#include "scap_machine_info.h"
+#include "scap_os_machine_info.h"
+#include "scap_limits.h"
+#include "scap_assert.h"
+#include "scap.h"
+#include "gettimeofday.h"
+
+#include <stdbool.h>
+#include <stdio.h>
+#include <windows.h>
+
+#define MSEC_TO_NS 1000000
+
+// https://stackoverflow.com/questions/10853985/programmatically-getting-system-boot-up-time-in-c-windows
+static uint64_t scap_windows_get_host_boot_time_ns()
+{
+	return GetTickCount64() * MSEC_TO_NS;
+}
+
+void scap_os_get_agent_info(scap_agent_info* agent_info)
+{
+	agent_info->start_ts_epoch = 0;
+	agent_info->start_time = 0;
+
+	/* Info 1:
+	 *
+	 * unix time in nsec of our startup time
+	 */
+	{
+		FILETIME creation_time, exit_time, kernel_time, user_time;
+		if(GetProcessTimes(GetCurrentProcess(), &creation_time, &exit_time, &kernel_time, &user_time))
+		{
+			agent_info->start_ts_epoch = ft_to_epoch_nsec(&creation_time);
+		}
+	}
+
+	/* Info 2:
+	 *
+	 * our startup time in seconds since boot
+	 */
+	if(agent_info->start_ts_epoch != 0)
+	{
+		uint64_t boot_time_ns = scap_windows_get_host_boot_time_ns();
+		agent_info->start_time = (agent_info->start_ts_epoch - boot_time_ns) / (1.0 * SECOND_TO_NS);
+	}
+
+	/* Info 3:
+	 *
+	 * Kernel release `uname -r` of the machine the agent is running on.
+	 */
+	{
+		OSVERSIONINFOA os_info;
+		os_info.dwOSVersionInfoSize = sizeof(OSVERSIONINFO);
+		if(GetVersionExA(&os_info))
```

Comment:
Note that this [probably won't return the correct version](https://learn.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getversionexa?redirectedfrom=MSDN) on modern versions of Windows. We use RtlGetVersion in Wireshark, but it looks like you can also call GetFileVersionInfo on one of the kernel DLLs.

Reply:
I'm happy it compiled without me seeing a Windows machine ;) I'm even happier to replace it with something that actually works.

For future me: https://github.com/wireshark/wireshark/blob/dcc9cbffefe0598eadec11cbc1869943a26219c5/wsutil/os_version_info.c#L192

(I don't think we need anything *that* fancy but I'll use it for inspiration ;])

Reply:
@incertum I think this falls in the "future PR" category.

Reply:
@gnosek I'll re-approve tomorrow as we need a rebase now!